MediaBrowser.MediaEncoding/Subtitles/SubtitleEncoder.cs

```diff
@@ -441,33 +455,47 @@ private string GetTextSubtitleFormat(MediaStream subtitleStream)
             }
         }
 
+        private string GetExportableSubtitleFileExtension(MediaStream subtitleStream)
+        {
+            // Using .pgssub as file extension is not allowed by ffmpeg. The file extension for pgs subtitles is .sup (or .sub).
+            if (string.Equals(subtitleStream.Codec, "pgssub", StringComparison.OrdinalIgnoreCase))
+            {
+                return "sup";
+            }
+            else
+            {
+                return GetExportableSubtitleFormat(subtitleStream);
+            }
+        }
+
         private bool IsCodecCopyable(string codec)
         {
             return string.Equals(codec, "ass", StringComparison.OrdinalIgnoreCase)
                 || string.Equals(codec, "ssa", StringComparison.OrdinalIgnoreCase)
                 || string.Equals(codec, "srt", StringComparison.OrdinalIgnoreCase)
-                || string.Equals(codec, "subrip", StringComparison.OrdinalIgnoreCase);
+                || string.Equals(codec, "subrip", StringComparison.OrdinalIgnoreCase)
+                || string.Equals(codec, "pgssub", StringComparison.OrdinalIgnoreCase);
         }
 
         /// <summary>
-        /// Extracts all text subtitles.
+        /// Extracts all extractable subtitles (text and pgs).
         /// </summary>
         /// <param name="mediaSource">The mediaSource.</param>
         /// <param name="cancellationToken">The cancellation token.</param>
         /// <returns>Task.</returns>
-        private async Task ExtractAllTextSubtitles(MediaSourceInfo mediaSource, CancellationToken cancellationToken)
+        private async Task ExtractAllExtractableSubtitles(MediaSourceInfo mediaSource, CancellationToken cancellationToken)
```

Comment:
Also rename the `ExtractAllTextSubtitlesInternal` to match `ExtractAllExtractableSubtitles`.